### PR TITLE
Adds tincan the robotics pet

### PR DESCRIPTION
**Pull request by BananaCreamPie**
tincan

Adds a custom sprited repair drone to robotics as an npc pet. His name is tincan. He says adorable things and zooms about.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/tincan.dm [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/mob/living/simple_animal/pet/drone/tincan/proc/wuv(change, mob/M)
+	if(change)
+		if(change > 0)
+			flick_overlay(image('icons/mob/animal.dmi',src,"heart-ani3",MOB_LAYER+1), list(M.client), 20)
+			emote("me", 1, "whirs around excitedly!")
+		if(change < 1)
+			emote("me", -1, "buzzes angrily.")
+
+
+/mob/living/simple_animal/pet/drone/tincan/movement_delay()
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/tincan.dm]: Are you trying to trigger the movement animation here? if so this is not the right way to do it.
Movement states in byond need 0 extra code.

Simple name the movement state the same as the normal one, and then tick it as a "Movement state" inside the .dmi, and byond will handle the rest.

You can then remove this block of code.

**Comment by Boggart** [comment]
Unless I'm mistaken that's like the little heart thing Ian does when you pet him.

**Comment by BananaCreamPie** [comment]
Its actually a custom made one for robotics to match :)

File: code/modules/mob/living/simple_animal/friendly/tincan.dm (hunk 1 above)
Comment (BananaCreamPie) [inline code/modules/mob/living/simple_animal/friendly/tincan.dm]: My bad I mustve left it in there from when I was playing around earlier. Removed now.

**Comment by Nienhaus** [comment]
:+1:

**Comment by RemieRichards** [comment]
Did you rename "drone_tincan_move" to "drone_tincan"? for movement states to work they have to match the default state name and be flagged as a Movementstate in the DMI

File: html/changelogs/bananacreampie-tincan.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, this gets changed to [] after reading.  Just remove the brackets when you add new shit.
+# Please surround your changes in  double quotes ("). It works without them, but if you use certain characters it screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - rscadd: "Adds an npc custom drone as robotics new pet 'tincan'
```
Comment (optimumtact) [inline html/changelogs/bananacreampie-tincan.yml]: Missing the ending " on this line

**Comment by GunHog** [comment]
Lovely little bugger! :+1:

**Comment by bgobandit** [comment]
can we get the heart mechanic for runtime

**Comment by Dorsisdwarf** [comment]
No, because cats dont love you enough.

**Comment by duncathan** [comment]
Isn't it against a drone's laws to love someone

File: code/modules/mob/living/simple_animal/friendly/tincan.dm [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+//cute hearts for adorable drone pet
+/mob/living/simple_animal/pet/drone/tincan/attack_hand(mob/living/carbon/human/M, mob/user)
+	. = ..()
+	switch(M.a_intent)
+		if("help")	wuv(1,M)
+		if("harm")	wuv(-1,M)
+
+/mob/living/simple_animal/pet/drone/tincan/proc/wuv(change, mob/M)
+	if(change)
```
Comment (phil235) [inline code/modules/mob/living/simple_animal/friendly/tincan.dm]: This needs an if(!stat) check.

**Comment by phil235** [comment]
What's the drone_tincan_hat sprite for?

**Comment by duncathan** [comment]
I imagine it would be for when you wear him as a hat.

**Comment by phil235** [comment]
merge conflicts

**Comment by Razharas** [comment]
@BananaCreamPie fix the conflicts and phils comment

**Comment by Razharas** [comment]
No reaction for over a week + no activity for a few weeks before that - ask if you want it reopened

**Pull request closed without merging**